R/create_default_parameters.R

```diff
@@ -0,0 +1,584 @@
+  # TODO: remove this section. User should update index_uncertainty from data
+  # object, not this default list.
+  index_uncertainty <- data@data |>
+    subset(name == fleet_name) |>
+    subset(type %in% c("landings", "index")) |>
+    (\(x) x$uncertainty)()
+
+  index_distribution_default <- switch(index_distribution,
+    "DnormDistribution" = create_default_DnormDistribution(
+      value = index_uncertainty, input_type = "data", data = data
+    ),
+    "DlnormDistribution" = create_default_DlnormDistribution(
+      value = index_uncertainty, input_type = "data", data = data
+    )
+  )
+  names(index_distribution_default) <- paste0(index_distribution, ".", names(index_distribution_default))
+
+  # Compile all default parameters into a single list
+  default <- list(c(selectivity_default, process_default, index_distribution_default))
+
+  names(default) <- fleet_name
+  return(default)
+}
+
+#' Create Default Maturity Parameters
+#'
+#' @description
+#' This function creates default parameters for maturity based on the specified
+#' form (e.g., LogisticMaturity).
+#' @param form A character. The form of maturity (e.g., LogisticMaturity).
+#' @return A list containing the default maturity parameters.
+#' @noRd
+create_default_maturity <- function(form = c("LogisticMaturity")) {
+  # Default parameters setup
+  # Currently only supports "LogisticMaturity", but can be extended
+  default <- list(switch(form,
+    "LogisticMaturity" = create_default_Logistic(),
+    check_valid_input(input = form, valid_options = c("LogisticMaturity"), arg_name = "form")
+  ))
+
+  names(default) <- "maturity"
+  names(default[["maturity"]]) <- paste0(form, ".", names(default[["maturity"]]))
+
+  return(default)
+}
+
+#' Create Default Beverton-Holt Recruitment parameters
+#'
+#' @description
+#' This function sets up default parameters for Beverton-Holt recruitment parameters.
+#' @param data An S4 object. FIMS input data.
+#' @return A list containing default recruitment parameters.
+#' @noRd
+create_default_BevertonHoltRecruitment <- function(data) {
+  # Check if data is an object from FIMSFrame class
+  check_arg_data(data)
+
+  # Create default parameters for Beverton-Holt recruitment
+  default <- list(
+    log_rzero.value = log(1e+06),
+    log_rzero.estimated = TRUE,
+    logit_steep.value = -log(1.0 - 0.75) + log(0.75 - 0.2),
+    logit_steep.estimated = FALSE,
+    log_devs.value = rep(0.0, data@n_years - 1),
+    log_devs.estimated = TRUE,
+    estimate_log_devs = TRUE
+  )
+  return(default)
+}
+
+#' Create Default DnormDistribution Parameters
+#'
+#' @description
+#' Create default parameters for DnormDistribution.
+#' @param value Default value for `log_sd`.
+#' @param data An S4 object. FIMS input data.
+#' @param input_type A character. Specifies if input is data or process.
+#' @return A list of default parameters for DnormDistribution.
+#' @noRd
+create_default_DnormDistribution <- function(value = log(0.4), data, input_type = "data") {
```

Comment:
The default here for value is `log(0.4)` but the only place that this function is used where value is not based on the uncertainty in the data is for the Beverton--Holt recruitment initialization. And there, the value is set to 0.1. I think that we should just be using the default in the Beverton--Holt recruitment initialization. @Bai-Li-NOAA were these values chosen with something in mind? Or, do you think it would be okay to use the default in all instances of the source code unless the values are based on data uncertainty. And, if so, what should the default be?

Reply:
Great point, I think we can use the BH recruitment initialization value here for consistency. In the future, it's fine to use the default unless the values are based on data uncertainty.